omnigibson/prims/entity_prim.py

```diff
@@ -207,22 +198,32 @@ def update_links(self, load_config=None):
                 # For cloth object, process prims that are Meshes
                 link_cls = ClothPrim
 
+            # Keep track of all the links we will create. We can't create that just yet because we need to find
+            # the base link first.
             if link_cls is not None:
-                self._links[link_name] = link_cls(
-                    prim_path=prim.GetPrimPath().__str__(),
-                    name=f"{self._name}:{link_name}",
-                    load_config=load_config,
-                )
+                links_to_create[link_name] = (link_cls, prim)
 
         # Infer the correct root link name -- this corresponds to whatever link does not have any joint existing
         # in the children joints
-        valid_root_links = list(set(self._links.keys()) - joint_children)
+        valid_root_links = list(set(links_to_create.keys()) - joint_children)
 
-        # TODO: Uncomment safety check here after we figure out how to handle legacy multi-bodied assets like bed with pillow
-        # assert len(valid_root_links) == 1, f"Only a single root link should have been found for this entity prim, " \
-        #                                    f"but found multiple instead: {valid_root_links}"
+        assert len(valid_root_links) == 1, f"Only a single root link should have been found for {self.name}, " \
```

Comment:
so just to clarify, we no longer have the bed + pillows multi-body-but-single-object case like in ig2?

Reply:
We do not, correct.